### PR TITLE
feature(url-sync): sort url parts. Fix #128

### LINKED ISSUE #128: [urlSync] order of url parameter
Would it be possible to keep the order in trackedParameters when hashing the current state ?

### PR DESCRIPTION
**Pull request by bobylito**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,8 +49,8 @@
   },
   "dependencies": {
     "algoliasearch": "^3.8.0",
-    "algoliasearch-helper": "^2.4.0",
-    "classnames": "^2.1.5",
+    "algoliasearch-helper": "^2.5.0",
+    "classnames": "^2.1.3",
```
Comment (redox) [inline package.json]: Is the `classnames` downgrade intended?

**Comment by redox** [comment]
Rebase + `npm shrinkwrap` = pain...

File: package.json (hunk 1 above)
Comment (bobylito) [inline package.json]: Dunno why but seems like it was needed so I did it. I do as NPM wants ;P

**Comment by bobylito** [comment]
No need to rebase if it can be merged, no?

File: npm-shrinkwrap.json [hunk 2]
```diff
@@ -244,7 +244,7 @@
     },
     "react-nouislider": {
       "version": "1.4.3",
-      "from": "react-nouislider@>=1.4.3 <2.0.0",
+      "from": "react-nouislider@>=1.3.0 <2.0.0",
```
Comment (vvo) [inline npm-shrinkwrap.json]: Woopsie, do not merge :D

Comment (vvo) [inline npm-shrinkwrap.json]: We need 1.4.3 here

File: package.json (hunk 1 above)
Comment (vvo) [inline package.json]: Should not be needed. NPM should not update this BUT if you ask it?

Comment (bobylito) [inline package.json]: Well the new version of the helper is a must so I need to update this !#(R$&$ shrinkwrap ;P

**Comment by bobylito** [comment]
Seems like I need to work on that a little bit more.

File: package.json (hunk 1 above)
Comment (vvo) [inline package.json]: Yes but here it's the classname dependency from package.json that was updated, I do not get why you would need to update the classname dependency. It's not used by the helper. You can just ignore the change by editing the commit, should be fine.

Comment (bobylito) [inline package.json]: I wasn't able to generate the new shrinkwrap in the current state. I must have done something wrong...

**Comment by pixelastic** [comment]
> No need to rebase if it can be merged, no?

If it can be merged, it can be rebased :). Ideally it would be better to rebase every PR, but this requires to switch back to cli, force push and then accept the pr (there's an issue [about that already](https://github.com/isaacs/github/issues/88)).

**Comment by bobylito** [comment]
if it's only commit that updates the version, it can be done again. Done it with NPM2 this time.

**Comment by vvo** [comment]
Yup seems good now, all the changes about resolved/.. are npm weirdness they try to fix since 2 years

**Pull request merged**